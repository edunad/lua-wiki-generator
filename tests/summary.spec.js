#!/usr/bin/env node
'use strict';

const fs = require('fs-extra');
const { resolve } = require('path');

const SummaryGenerator = require('../src/summary.js');

const SUMMARY_TEMPLATE_FILE = './__test_templates__/SUMMARY_TEMPLATE.md';
const INPUT_TEST_FOLDER = './tests/__test_summary__';

describe('SummaryGenerator', () => {
    beforeAll(async () => {
        this.summary = await fs.readFile(resolve(__dirname, SUMMARY_TEMPLATE_FILE), 'utf8');
    });

    beforeEach(() => {
        SummaryGenerator.setLinkMDParser(null);
    });

    it('Generates summary', async () => {
        const generated = await SummaryGenerator.generate(
            INPUT_TEST_FOLDER,
            {
                './tests/__test_summary__/io/io_open.md': {
                    title: { msg: 'bad_class_test:test' },
                    type: 'METHOD',
                },
                './tests/__test_summary__/console/console_bottom.md': {
                    title: { msg: 'bad_class_test:test' },
                    type: 'METHOD',
                },
            },
            this.summary,
        );

        expect(generated).toMatchSnapshot();
    });

    it('Generates summary with a custom link', async () => {
        SummaryGenerator.setLinkMDParser((type, linkMap, data) => {
            if (type === 'SUMMARY') {
                if (data.fileName) {
                    // Not root
                    return `${data.fileName}`;
                } else {
                    // ROOT
                    return `${data.dir}/MY_ROOT_DOCUMENT.md`;
                }
            }

            throw new Error(`Unknown type: ${type}`);
        });

        const generated = await SummaryGenerator.generate(
            INPUT_TEST_FOLDER,
            {
                './tests/__test_summary__/io/io_open.md': {
                    title: { msg: 'bad_class_test:test' },
                    type: 'METHOD',
                },
                './tests/__test_summary__/console/console_bottom.md': {
                    title: { msg: 'bad_class_test:test' },
                    type: 'METHOD',
                },
            },
            this.summary,
        );

        expect(generated).toMatchSnapshot();
    });
});
