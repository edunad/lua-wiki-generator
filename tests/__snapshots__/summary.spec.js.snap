// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryGenerator Generates summary 1`] = `
"# Table of contents

* [☄ Welcome to the wiki!](README.md)
  * [console](./tests/__test_summary__/console/README.md)
    * [bad_class_test:test](./tests/__test_summary__/console/console_bottom.md)
  * [io](./tests/__test_summary__/io/README.md)
    * [bad_class_test:test](./tests/__test_summary__/io/io_open.md)
"
`;

exports[`SummaryGenerator Generates summary with a custom link 1`] = `
"# Table of contents

* [☄ Welcome to the wiki!](README.md)
  * ./tests/__test_summary__/console/MY_ROOT_DOCUMENT.md
    * console_bottom.md
  * ./tests/__test_summary__/io/MY_ROOT_DOCUMENT.md
    * io_open.md
"
`;
